Paltineanu Rares-Mihai/3.7 - curs4 switch/main.c

```diff
@@ -0,0 +1,112 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <math.h>
+
+int verif(char hex[200])
+{
+    int i = strlen(hex) - 1;
+    for(int j = 0; j <= i; j++)
+    {
+        if(hex[j]>'F')
+        {
+            return 0;
+        }
+
+    }
+    return 1;
+}
+
+///firstly i write the function like this: void convert(char hex[200], double &a)
+///but the error is saying "expected ';', ',', or ')' before '&' token and idk what is meaning
+
+int convert(char hex[200], double a)
+{
+    int b = strlen(hex);
+    for(int i = 0; i < b; i++)
+    {
+        switch(hex[i])
+        {
+        case '1':
+            a = a + (hex[i] - '0')*(pow(16,strlen(hex) - i - 1));
```

Comment:
These lines are all the same, I'm not sure if `C` compilers will let you write what I'm about to show you, but you can try:

```
case '1':
case '2':
case '3':
case '4':
case '5':
case '6':
case '7':
case '8':
case '9':
    a = a + (hex[i] - '0')*(pow(16,strlen(hex) - i - 1));
    break
```

Reply:
Again, it might not work in C but it's worth the shot

Reply:
I tried and it worked!

Reply:
i tried and it worked